library/ssl_srv.c

```diff
@@ -116,6 +116,11 @@ static int ssl_parse_servername_ext( mbedtls_ssl_context *ssl,
 
         if( p[0] == MBEDTLS_TLS_EXT_SERVERNAME_HOSTNAME )
         {
+            if( mbedtls_ssl_parse_dns( p + 3, hostname_len ) != 0 )
```

Comment:
At the moment we do not really want to change the behaviour of the library, just report in the debug log that the ServerName is incorrect.